make libquip.a fails on GCC-14 (darwin_arm64_gfortran)

## Problem
I have noticed that I can't build that public branch with gcc-14 on an ARM Mac. Tried on commit 7c428af0fb8b60f73051720c24e9a40ee5484a1c

Compiler & OS versions: 
- `gcc (Homebrew GCC 14.1.0_2) 14.1.0`
- `GNU Fortran (Homebrew GCC 14.1.0_2) 14.1.0`
- MacOS 14.5 Sonoma

## My (unsafe?) patch locally

If I add ` -Wno-error=return-mismatch` to the `CFLAGS` then this goes away, but compiling everything with that setting may not be the wisest solution. What do you think we should do?


## Error log

```
$make libquip.a                                                                                                                                                                                                                                
```

```
Using QUIP_ARCH=darwin_arm64_gfortran, MODULES=libAtoms GAP Potentials Utils, QUIP_ROOT=/Users/tks32/research/castep/opt/QUIP
********************************************

 Making libAtoms 

********************************************
rm -f /Users/tks32/research/castep/opt/QUIP/build/darwin_arm64_gfortran/Makefile
cp /Users/tks32/research/castep/opt/QUIP/src/libAtoms/Makefile /Users/tks32/research/castep/opt/QUIP/build/darwin_arm64_gfortran/Makefile
make -C /Users/tks32/research/castep/opt/QUIP/build/darwin_arm64_gfortran QUIP_ROOT=/Users/tks32/research/castep/opt/QUIP VPATH=/Users/tks32/research/castep/opt/QUIP/src/libAtoms -I/Users/tks32/research/castep/opt/QUIP -I/Users/tks32/research/castep/opt/QUIP/arch
make[1]: Entering directory '/Users/tks32/research/castep/opt/QUIP/build/darwin_arm64_gfortran'
gcc -fPIC -I/Users/tks32/research/castep/opt/QUIP/src/libAtoms -I/Users/tks32/research/castep/opt/QUIP/src/fox/objs.darwin_arm64_gfortran/finclude -O3  -DFORTRAN_UNDERSCORE  -DGETARG_F2003 -DGETENV_F2003 -DGFORTRAN -DDARWIN -fPIC -D'GIT_VERSION="https://github.com/libAtoms/QUIP,7c428af-dirty"' -D'GAP_VERSION=1686220958' -D'QUIP_ARCH="darwin_arm64_gfortran"' -D'SIZEOF_FORTRAN_T=2' -DHAVE_GAP -DHAVE_PRECON -DHAVE_QR  -c /Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c
In file included from /Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:50:
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c: In function 'xyz_find_index':
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:180:5: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  180 |     RAISE_ERROR_WITH_KIND(ERROR_IO, "Cannot access xyz file %s\n", fname);
      |     ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:169:5: note: declared here
  169 | int xyz_find_index(char *fname, char *indexname, int *do_update, int *error) {
      |     ^~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:203:7: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  203 |       RAISE_ERROR_WITH_KIND(ERROR_IO, "Cannot stat xyz file %s\n", fname);
      |       ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:169:5: note: declared here
  169 | int xyz_find_index(char *fname, char *indexname, int *do_update, int *error) {
      |     ^~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:206:7: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  206 |       RAISE_ERROR_WITH_KIND(ERROR_IO, "Cannot stat xyz.idx file %s\n", fname);
      |       ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:169:5: note: declared here
  169 | int xyz_find_index(char *fname, char *indexname, int *do_update, int *error) {
      |     ^~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c: In function 'xyz_read_index':
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:224:5: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  224 |     RAISE_ERROR_WITH_KIND(ERROR_IO, "Index file %s cannot be opened\n", indexname);
      |     ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:214:5: note: declared here
  214 | int xyz_read_index(char *indexname, long **frames, int **atoms, int *frames_array_size, int *error) {
      |     ^~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:227:5: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  227 |     RAISE_ERROR_WITH_KIND(ERROR_IO, "Index file %s is empty\n",indexname);
      |     ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:214:5: note: declared here
  214 | int xyz_read_index(char *indexname, long **frames, int **atoms, int *frames_array_size, int *error) {
      |     ^~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:233:7: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  233 |       RAISE_ERROR_WITH_KIND(ERROR_IO, "Premature end of indexfile %s\n",indexname);
      |       ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:214:5: note: declared here
  214 | int xyz_read_index(char *indexname, long **frames, int **atoms, int *frames_array_size, int *error) {
      |     ^~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c: In function 'xyz_update_index':
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:250:5: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  250 |     RAISE_ERROR_WITH_KIND(ERROR_IO, "xyz_update_index: cannot open %s for reading", fname);
      |     ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:243:5: note: declared here
  243 | int xyz_update_index(char *fname, char *indexname, long **frames, int **atoms, int *frames_array_size, int nframes, int *error) {
      |     ^~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:281:7: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  281 |       RAISE_ERROR_WITH_KIND(ERROR_IO, "xyz_find_frames: malformed XYZ file %s at frame %d\n",fname,nframes);
      |       ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:243:5: note: declared here
  243 | int xyz_update_index(char *fname, char *indexname, long **frames, int **atoms, int *frames_array_size, int nframes, int *error) {
      |     ^~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c: In function 'xyz_find_frames':
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:128:155: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  128 | #define PASS_ERROR if (error != NULL && *error != ERROR_NONE) { error_h_line = __LINE__; c_push_error_(__FILE__, &error_h_line, error, strlen(__FILE__)); return; }
      |                                                                                                                                                           ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:346:3: note: in expansion of macro 'PASS_ERROR'
  346 |   PASS_ERROR;
      |   ^~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:337:5: note: declared here
  337 | int xyz_find_frames(char *fname, long **frames, int **atoms, int *frames_array_size, int *error) {
      |     ^~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:128:155: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  128 | #define PASS_ERROR if (error != NULL && *error != ERROR_NONE) { error_h_line = __LINE__; c_push_error_(__FILE__, &error_h_line, error, strlen(__FILE__)); return; }
      |                                                                                                                                                           ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:351:5: note: in expansion of macro 'PASS_ERROR'
  351 |     PASS_ERROR;
      |     ^~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:337:5: note: declared here
  337 | int xyz_find_frames(char *fname, long **frames, int **atoms, int *frames_array_size, int *error) {
      |     ^~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:128:155: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  128 | #define PASS_ERROR if (error != NULL && *error != ERROR_NONE) { error_h_line = __LINE__; c_push_error_(__FILE__, &error_h_line, error, strlen(__FILE__)); return; }
      |                                                                                                                                                           ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:356:5: note: in expansion of macro 'PASS_ERROR'
  356 |     PASS_ERROR;
      |     ^~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:337:5: note: declared here
  337 | int xyz_find_frames(char *fname, long **frames, int **atoms, int *frames_array_size, int *error) {
      |     ^~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:126:300: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  126 | #define RAISE_ERROR(info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = ERROR_UNSPECIFIED; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                            ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:359:7: note: in expansion of macro 'RAISE_ERROR'
  359 |       RAISE_ERROR("xyz_find_frames: empty file!");
      |       ^~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:337:5: note: declared here
  337 | int xyz_find_frames(char *fname, long **frames, int **atoms, int *frames_array_size, int *error) {
      |     ^~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:128:155: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  128 | #define PASS_ERROR if (error != NULL && *error != ERROR_NONE) { error_h_line = __LINE__; c_push_error_(__FILE__, &error_h_line, error, strlen(__FILE__)); return; }
      |                                                                                                                                                           ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:363:5: note: in expansion of macro 'PASS_ERROR'
  363 |     PASS_ERROR;
      |     ^~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:337:5: note: declared here
  337 | int xyz_find_frames(char *fname, long **frames, int **atoms, int *frames_array_size, int *error) {
      |     ^~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c: In function 'get_line':
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:413:7: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  413 |       RAISE_ERROR_WITH_KIND(ERROR_IO_EOF, info);
      |       ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:405:7: note: declared here
  405 | char* get_line(char *linebuffer, int string, int string_length, char *orig_stringp, char *stringp, char **prev_stringp,
      |       ^~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:425:9: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  425 |         RAISE_ERROR_WITH_KIND(ERROR_IO, "cannot strip prefix from line <%s>", linebuffer);
      |         ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:405:7: note: declared here
  405 | char* get_line(char *linebuffer, int string, int string_length, char *orig_stringp, char *stringp, char **prev_stringp,
      |       ^~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:431:7: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  431 |       RAISE_ERROR_WITH_KIND(ERROR_IO_EOF, info);
      |       ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:405:7: note: declared here
  405 | char* get_line(char *linebuffer, int string, int string_length, char *orig_stringp, char *stringp, char **prev_stringp,
      |       ^~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/libatoms.h:127:303: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  127 | #define RAISE_ERROR_WITH_KIND(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return; } else c_error_abort_(error)
      |                                                                                                                                                                                                                                                                                                               ^~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:438:9: note: in expansion of macro 'RAISE_ERROR_WITH_KIND'
  438 |         RAISE_ERROR_WITH_KIND(ERROR_IO, "cannot strip prefix from line <%s>", linebuffer);
      |         ^~~~~~~~~~~~~~~~~~~~~
/Users/tks32/research/castep/opt/QUIP/src/libAtoms/xyz.c:405:7: note: declared here
  405 | char* get_line(char *linebuffer, int string, int string_length, char *orig_stringp, char *stringp, char **prev_stringp,
      |       ^~~~~~~~
make[1]: *** [Makefile:141: xyz.o] Error 1
make[1]: Leaving directory '/Users/tks32/research/castep/opt/QUIP/build/darwin_arm64_gfortran'
make: *** [Makefile:155: libAtoms] Error 2
~/research/castep/opt/QUIP public +1 !1 ❯                                                                                                                                                                                                                                               05:52:44 pm


```

--- a/src/libAtoms/libatoms.h
+++ b/src/libAtoms/libatoms.h
@@ -128,6 +128,11 @@
 #define PASS_ERROR if (error != NULL && *error != ERROR_NONE) { error_h_line = __LINE__; c_push_error_(__FILE__, &error_h_line, error, strlen(__FILE__)); return; }
 #define CLEAR_ERROR c_error_clear_stack_();
 
+#define RAISE_ERROR_INT(info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = ERROR_UNSPECIFIED; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return -1; } else c_error_abort_(error)
+#define RAISE_ERROR_WITH_KIND_INT(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return -1; } else c_error_abort_(error)
+#define PASS_ERROR_INT if (error != NULL && *error != ERROR_NONE) { error_h_line = __LINE__; c_push_error_(__FILE__, &error_h_line, error, strlen(__FILE__)); return -1; }
+#define RAISE_ERROR_WITH_KIND_CHAR(kind, info, ...) sprintf(error_h_info, info, ## __VA_ARGS__ ); error_h_line = __LINE__; error_h_kind = kind; c_push_error_with_info_(error_h_info, __FILE__, &error_h_line, &error_h_kind, strlen(error_h_info), strlen(__FILE__)); if (error != NULL) { *error = error_h_kind; return '\0'; } else c_error_abort_(error)
+
 extern void c_push_error_with_info_(char*, char*, int*, int*, size_t, size_t);
 extern void c_push_error_(char*, int*, int*, size_t);
 extern void c_error_abort_(int *);

--- a/src/libAtoms/xyz.c
+++ b/src/libAtoms/xyz.c
@@ -177,7 +177,7 @@ int xyz_find_index(char *fname, char *indexname, int *do_update, int *error) {
   strncat(indexname, ".idx", LINESIZE-strlen(indexname)-1);
 
   if (access(fname, R_OK) != 0) {
-    RAISE_ERROR_WITH_KIND(ERROR_IO, "Cannot access xyz file %s\n", fname);
+    RAISE_ERROR_WITH_KIND_INT(ERROR_IO, "Cannot access xyz file %s\n", fname);
   }
 
   idx_exists = access(indexname, R_OK) == 0;
@@ -200,10 +200,10 @@ int xyz_find_index(char *fname, char *indexname, int *do_update, int *error) {
   *do_update = 1;
   if (idx_exists) {
     if (stat(fname, &xyz_stat) != 0) {
-      RAISE_ERROR_WITH_KIND(ERROR_IO, "Cannot stat xyz file %s\n", fname);
+      RAISE_ERROR_WITH_KIND_INT(ERROR_IO, "Cannot stat xyz file %s\n", fname);
     }
     if (stat(indexname, &idx_stat) != 0) {
-      RAISE_ERROR_WITH_KIND(ERROR_IO, "Cannot stat xyz.idx file %s\n", fname);
+      RAISE_ERROR_WITH_KIND_INT(ERROR_IO, "Cannot stat xyz.idx file %s\n", fname);
     }
     *do_update = xyz_stat.st_mtime > idx_stat.st_mtime;
   }
@@ -221,16 +221,16 @@ int xyz_read_index(char *indexname, long **frames, int **atoms, int *frames_arra
   debug("xyz_read_index: reading XYZ index from file %s\n", indexname);
   index = fopen(indexname, "r");
   if (index == NULL) {
-    RAISE_ERROR_WITH_KIND(ERROR_IO, "Index file %s cannot be opened\n", indexname);
+    RAISE_ERROR_WITH_KIND_INT(ERROR_IO, "Index file %s cannot be opened\n", indexname);
   }
   if (!fgets(linebuffer,LINESIZE,index)) {
-    RAISE_ERROR_WITH_KIND(ERROR_IO, "Index file %s is empty\n",indexname);
+    RAISE_ERROR_WITH_KIND_INT(ERROR_IO, "Index file %s is empty\n",indexname);
   }
   sscanf(linebuffer, "%d", &nframes);
   realloc_frames(frames, atoms, frames_array_size, nframes+2);
   for (i=0; i<=nframes; i++) {
     if (!fgets(linebuffer,LINESIZE,index)) {
-      RAISE_ERROR_WITH_KIND(ERROR_IO, "Premature end of indexfile %s\n",indexname);
+      RAISE_ERROR_WITH_KIND_INT(ERROR_IO, "Premature end of indexfile %s\n",indexname);
     }
     sscanf(linebuffer, "%ld %d", &(*frames)[i], &(*atoms)[i]);
     debug("index %ld %d\n", (*frames)[i], (*atoms)[i]);
@@ -247,7 +247,7 @@ int xyz_update_index(char *fname, char *indexname, long **frames, int **atoms, i
 
   in = fopen(fname, "r");
   if (in == NULL) {
-    RAISE_ERROR_WITH_KIND(ERROR_IO, "xyz_update_index: cannot open %s for reading", fname);
+    RAISE_ERROR_WITH_KIND_INT(ERROR_IO, "xyz_update_index: cannot open %s for reading", fname);
   }
 
   if (nframes != 0) {
@@ -278,7 +278,7 @@ int xyz_update_index(char *fname, char *indexname, long **frames, int **atoms, i
     realloc_frames(frames, atoms, frames_array_size, nframes+2);
     (*frames)[nframes] = ftell(in)-strlen(linebuffer);
     if (sscanf(linebuffer, "%d", &natoms) != 1) {
-      RAISE_ERROR_WITH_KIND(ERROR_IO, "xyz_find_frames: malformed XYZ file %s at frame %d\n",fname,nframes);
+      RAISE_ERROR_WITH_KIND_INT(ERROR_IO, "xyz_find_frames: malformed XYZ file %s at frame %d\n",fname,nframes);
     }
 
     (*atoms)[nframes] = natoms;
@@ -343,24 +343,24 @@ int xyz_find_frames(char *fname, long **frames, int **atoms, int *frames_array_s
   INIT_ERROR;
 
   got_index = xyz_find_index(fname, indexname, &do_update, error);
-  PASS_ERROR;
+  PASS_ERROR_INT;
 
   nframes = 0;
   if (got_index) {
     nframes = xyz_read_index(indexname, frames, atoms, frames_array_size, error);
-    PASS_ERROR;
+    PASS_ERROR_INT;
   }
 
   if (!got_index || do_update) {
     nframes = xyz_update_index(fname, indexname, frames, atoms, frames_array_size, nframes, error);
-    PASS_ERROR;
+    PASS_ERROR_INT;
 
     if (nframes == 0) {
-      RAISE_ERROR("xyz_find_frames: empty file!");
+      RAISE_ERROR_INT("xyz_find_frames: empty file!");
     }
 
     xyz_write_index(indexname, frames, atoms, frames_array_size, nframes, error);
-    PASS_ERROR;
+    PASS_ERROR_INT;
   }
 
   debug("xyz_find_frames: %s: found %d complete frames\n", fname, nframes);
@@ -410,7 +410,7 @@ char* get_line(char *linebuffer, int string, int string_length, char *orig_strin
   *line_offset = 0;
   if (string) {
     if (*stringp == '\0' || (string_length != 0 && (stringp-orig_stringp >= string_length))) {
-      RAISE_ERROR_WITH_KIND(ERROR_IO_EOF, info);
+      RAISE_ERROR_WITH_KIND_CHAR(ERROR_IO_EOF, info);
     }
     *prev_stringp = stringp;
     while (*stringp != '\n' && *stringp != '\0' && (string_length == 0 || stringp-orig_stringp < string_length)) stringp++;
@@ -422,20 +422,20 @@ char* get_line(char *linebuffer, int string, int string_length, char *orig_strin
       if (strchr(linebuffer, ' ') != NULL) {
         *line_offset = strchr(linebuffer, ' ')+1-linebuffer;
       } else {
-        RAISE_ERROR_WITH_KIND(ERROR_IO, "cannot strip prefix from line <%s>", linebuffer);
+        RAISE_ERROR_WITH_KIND_CHAR(ERROR_IO, "cannot strip prefix from line <%s>", linebuffer);
       }
     }
     return stringp;
   } else {
     if (!fgets(linebuffer,LINESIZE,in)) {
-      RAISE_ERROR_WITH_KIND(ERROR_IO_EOF, info);
+      RAISE_ERROR_WITH_KIND_CHAR(ERROR_IO_EOF, info);
     }
     linebuffer[strlen(linebuffer)-1] = '\0';
     if (strip_prefix) {
       if (strchr(linebuffer, ' ') != NULL) {
         *line_offset = strchr(linebuffer, ' ')+1-linebuffer;
       } else {
-        RAISE_ERROR_WITH_KIND(ERROR_IO, "cannot strip prefix from line <%s>", linebuffer);
+        RAISE_ERROR_WITH_KIND_CHAR(ERROR_IO, "cannot strip prefix from line <%s>", linebuffer);
       }
       //debug("line = <%s>\n", linebuffer);
     }

